pkg/azure/keyvault/credentials.go

```diff
@@ -2,36 +2,99 @@ package keyvault
 
 import (
 	"os"
+	"strings"
 
 	"get.porter.sh/plugin/azure/pkg/azure/azureconfig"
 	"github.com/Azure/azure-sdk-for-go/services/keyvault/auth"
 	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
 	azureauth "github.com/Azure/go-autorest/autorest/azure/auth"
 	"github.com/hashicorp/go-hclog"
 	"github.com/pkg/errors"
 )
 
+// GetCredentials gets an authorizer for Azure
 func GetCredentials(cfg azureconfig.Config, l hclog.Logger) (autorest.Authorizer, error) {
-	prefix := cfg.EnvServicePrincipalPrefix
+
+	azureAuthEnvVarNames := []string{
+		azureauth.TenantID,
+		azureauth.ClientID,
+		azureauth.ClientSecret,
+		azureauth.CertificatePath,
+		azureauth.CertificatePassword,
+		azureauth.Username,
+		azureauth.Password,
+	}
+
+	prefix := cfg.EnvAzurePrefix
 	if prefix != "" && prefix != "AZURE_" {
-		tenantEnv := prefix + "TENANT_ID"
-		clientEnv := prefix + "CLIENT_ID"
-		secretEnv := prefix + "CLIENT_SECRET"
-
-		tenant := os.Getenv(tenantEnv)
-		client := os.Getenv(clientEnv)
-		secret := os.Getenv(secretEnv)
-
-		// NewAuthorizerFromEnvironment only reads from well-known env vars
-		os.Setenv(azureauth.TenantID, tenant)
-		os.Setenv(azureauth.ClientID, client)
-		os.Setenv(azureauth.ClientSecret, secret)
+		for _, v := range azureAuthEnvVarNames {
+			env := prefix + strings.TrimPrefix(v, "AZURE_")
+			val := os.Getenv(env)
+			os.Setenv(v, val)
+		}
 	}
 
-	authorizer, err := auth.NewAuthorizerFromEnvironment()
+	var authorizer autorest.Authorizer
+	var err error
+
+	// Attempt to login with az cli if no vars are set.
+
+	if noAzureAuthEnvVarsAreSet(azureAuthEnvVarNames) {
+		authorizer, err = auth.NewAuthorizerFromCLI()
+		if err != nil {
+			return nil, errors.Wrap(err, "Failed to create an azure authorizer from azure cli")
+		}
+
+		return authorizer, nil
+	}
+
+	// NewAuthorizierFromEnvironment attempts to authenticate using credentials, certicates, user name and password and MSI however if we get here MSI login wll be skipped as env vars are set so one of the other methods will be attempted
+
+	authorizer, err = auth.NewAuthorizerFromEnvironment()
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to create an azure authorizer")
+		return nil, errors.Wrap(err, "Failed to create an azure authorizer from environment")
 	}
 
 	return authorizer, nil
 }
+
+// getAzureKeyVaultResourceID gets the Key Vault endpoint resource
```

Comment:
This function doesn't appear to be referenced/used anywhere... is that correct?  Similarly, the only reference to `getAzureEnvironment` appears to be in this function... so, if not used, should they both be removed?

Reply:
Removed them, these are used by some additional auth options that I split out into another change that I havent submitted a PR for yet.